### PR TITLE
Next: New version with Nette v3

### LINKED ISSUE #25: Nette 3.0 compatiblity issue
There seem to be problem with usage of deprecated method in ConsoleExtension on line 127 as the method second argument (`setClass`) was deprecated.

Tested with `v3.0.0-beta2`

### PR DESCRIPTION
**Pull request by f3l1x**
- Closes #25 
- Require PHP ^7.2
- Test against PHP 7.2-8.0
- Support new yummy nette/schema. Refactor configuration to allow pass any key needed.
- Lazy by default.
- symfony/console >= 4.2.5.

----

```
{
  "require": {
    "contributte/console": "^0.7.0"
  }
}
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,62 +1,55 @@
...
-  # Print *.actual content
-  - for i in $(find tests -name \*.actual); do echo "--- $i"; cat $i; echo; echo; done
+  - for i in $(find tests -name \*.actual); do echo "--- $i"; cat $i; echo; echo; done # Print *.actual content
 
 jobs:
   include:
-    - env: title="Lowest Dependencies 7.1"
-      php: 7.1
+    - env: title="Lowest Dependencies 7.3"
+      php: 7.3
```
Comment (mabar) [inline .travis.yml]: Imho should be used lowest compatible php version to ensure features from hígher php versions are not required by code.

File: composer.json [hunk 2]
```diff
@@ -16,58 +16,41 @@
     }
   ],
   "require": {
-    "php": ">= 7.1",
-    "symfony/console": "^4.0.0"
+    "php": "^7.2",
```
Comment (mabar) [inline composer.json]: Nice idea, future-proof

File: composer.json [hunk 3]
```diff
@@ -16,58 +16,41 @@
...
+    "symfony/console": "^4.2.5"
   },
   "require-dev": {
-    "ninjify/qa": "^0.8.0",
-    "ninjify/nunjuck": "^0.2.0",
-    "nette/di": "~2.4.11",
-    "nette/http": "~2.4.8"
+    "ninjify/qa": "^0.9.0",
+    "ninjify/nunjuck": "^0.3.0",
+    "nette/di": "~3.0.0",
```
Comment (mabar) [inline composer.json]: Isn't DI the main and only feature of that package?

**Comment by JakubBarta** [comment]
Hi guys,
is there something else that has to be done before this PR is merged (and a version compatible with Nette 3.0 is released)?
If so, I'll gladly help

**Pull request merged**